units/pod_allocator.go

```diff
@@ -107,9 +107,37 @@ func (j *podAllocatorJob) Run(ctx context.Context) {
 			j.AddRetryableError(errors.Wrap(err, "marking unallocatable container task as system-failed"))
 		}
 
+		grip.Info(message.Fields{
+			"message": "refusing to allocate task because it has no more allocation attempts",
+			"task":    j.task.Id,
+			"project": j.pRef.Identifier,
+			"job":     j.ID(),
+		})
+
 		return
 	}
 	if !j.task.ShouldAllocateContainer() {
+		grip.Info(message.Fields{
+			"message": "refusing to allocate task because it is not in a valid state to be allocated",
+			"task":    j.task.Id,
+			"project": j.pRef.Identifier,
+			"job":     j.ID(),
+		})
+		return
+	}
+
+	canDispatch, reason := model.ProjectCanDispatchTask(j.pRef, j.task)
+	if !canDispatch {
+		if reason == "" {
```

Comment:
We can also default `reason` to be "of unknown reason" on L1 of `ProjectCanDispatchTask` if you think that's cleaner

Reply:
Hmm, I looked back at this logic and I think we don't need to deal with the empty string case anymore since it'll just show up in the log as an empty field. I removed this.